_overviews/scala3-book/scala-for-javascript-devs.md

```diff
@@ -0,0 +1,1238 @@
+  case Thin, Thick, Regular
+
+enum Topping:
+  case Cheese, Pepperoni, BlackOlives, GreenOlives, Onions
+
+// import those enumerations and the ArrayBuffer,
+// so the Pizza class can use them
+import CrustSize._
+import CrustType._
+import Topping._
+import scala.collection.mutable.ArrayBuffer
+
+// define an OOP style Pizza class
+class Pizza(
+  var crustSize: CrustSize,
+  var crustType: CrustType
+):
+
+  private val toppings = ArrayBuffer[Topping]()
+
+  def addTopping(t: Topping): Unit = 
+    toppings += t
+
+  def removeTopping(t: Topping): Unit = 
+    toppings -= t
+
+  def removeAllToppings(): Unit = 
+    toppings.clear()
+
+  override def toString(): String =
+    s"""
+      |Pizza:
+      |  Crust Size: ${crustSize}
+      |  Crust Type: ${crustType}
+      |  Toppings:   ${toppings}
+    """.stripMargin
+
+end Pizza
+
+// create a Pizza instance
+val p = Pizza(Small, Thin)
+
+// change the crust
+p.crustSize = Large
+p.crustType = Thick
+
+// add and remove toppings
+p.addTopping(Cheese)
+p.addTopping(Pepperoni)
+p.addTopping(BlackOlives)
+p.removeTopping(Pepperoni)
+
+// print the pizza, which uses its `toString` method
+println(p)
+```
+
+
+
+## Interfaces, traits, and inheritance
+
+Scala uses traits as interfaces, and also to create mixins. Traits can have both abstract and concrete members, including methods and fields.
+
+This example shows how to define two traits, create a class that extends and implements those traits, and then create and use an instance of that class:  
+
+```scala
+trait HasLegs:
+  def numLegs: Int
+  def walk(): Unit
+  def stop() = println("Stopped walking")
+
+trait HasTail:
+  def wagTail(): Unit
+  def stopTail(): Unit
+
+class Dog(var name: String) extends HasLegs, HasTail:
+  val numLegs = 4
+  def walk() = println("I’m walking")
+  def wagTail() = println("⎞⎜⎛  ⎞⎜⎛")
+  def stopTail() = println("Tail is stopped")
+  override def toString = s"$name is a Dog"
+
+// create a Dog instance
+val d = Dog("Rover")
+
+// use the class’s attributes and behaviors
+println(d.numLegs)   // 4
+d.wagTail()          // "⎞⎜⎛  ⎞⎜⎛"
+d.walk()             // "I’m walking"
+```
+
+
+
+## Control Structures 
+
+Except for the use of `===` and `!==` in JavaScript, comparison and logical operators are almost identical in JavaScript and Scala:  
+
+<table>
+  <tbody>
+    <tr>
+      <th valign="top">JavaScript</th>
+      <th valign="top">Scala</th>
+    </tr>
+    <tr>
+      <td rowspan="1" colspan="2" class="table-desc-row">Comparison operators</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>==</code></td>
+      <td valign="top"><code>==</code></td>
+    </tr>
+    <tr>
+      <td valign="top"><code>===</code></td>
+      <td valign="top"><code>==</code></td>
+    </tr>
+    <tr>
+      <td valign="top"><code>!=</code></td>
+      <td valign="top"><code>!=</code></td>
+    </tr>
+    <tr>
+      <td valign="top"><code>!==</code></td>
+      <td valign="top"><code>!=</code></td>
+    </tr>
```

Comment:
I suggested to @alvinj to list them not next to each other but one after another (below each other), this would also would make it less likely to suggest that they are equivalent.

Reply:
We are going to reformat this page (and the Python and Java) pages to show one example after the other. Is there any other text I should add here? I haven’t used plain JavaScript in a long time, and I didn’t know they are so dissimilar.